### PR TITLE
Adding in unk postprocessing code

### PR DESCRIPTION
**Pull request by annazuo**

### CODE REVIEW

File: decoder.py [hunk 1]
```diff
@@ -118,14 +119,30 @@ def forward(self, input, encoder_outs, hidden_init, targets_len):
             # print(f'log_prob shape {log_prob.shape}')
             # print(f'log_prob {log_prob}')
 
+            # FOR UNK POSTPROCESSING
```
Comment (jessbunnag) [inline decoder.py]: @annazuo thoughts on moving this out of decoder and instead make decoder's forward() return attention scores so you can compute `best_attn_labels`? my reasoning is that
1. we should separate the functionality of decoder and inference. technically post-processing is part of inference and happens after the model has been trained. (see section 4.3 of the paper)
2. if might be useful for plotting attention mask if we return the attention scores from the decoder too

File: train.ipynb [hunk 2]
```diff
@@ -131,7 +131,7 @@
     "    \n",
     "    model.train()\n",
     "\n",
-    "    dec_log_probs, dec_hidden = model(input, target, inputs_len, target_len)\n",
+    "    dec_log_probs, dec_hidden, best_attn_labels = model(input, target, inputs_len, target_len)\n",
```
Comment (jessbunnag) [inline train.ipynb]: as i said above, let's change this so the model returns attention scores instead and we implement post processing in the train loop. IMO for cleaner code and better debugging you can create `inference.py` and put a function that does post processing in there.

File: decoder.py [hunk 3]
```diff
@@ -118,14 +119,30 @@ def forward(self, input, encoder_outs, hidden_init, targets_len):
...
+            # first, get the src token index with the max attention for this time step
+            src_token_idxs_t = torch.argmax(attn_scores, dim=1)
+            # print(f'src_token_idxs_t shape {src_token_idxs_t.shape}')
+
+            # convert the index location in the src to the vocab class label for src
+            src_labels_t = []
+            for sen_idx in range(len(src_token_idxs_t)):
+                class_label = answer[sen_idx][src_token_idxs_t[sen_idx]]
```
Comment (jessbunnag) [inline decoder.py]: when you implement this outside of decoder in train loop, you can potentially use source_vocab instead

**Review by jessbunnag — CHANGES_REQUESTED** [review]
unk post processing looks sound to me. left comments about code organization
